### DISCUSSION TITLE
Using Defer helper in laravel 11

### DISCUSSION THREAD

**Discussion by MohamedAbdel-wahed** [laravel/framework#52863] [Q&A]
```
Route::get('/', function () {
    defer(function () {
        sleep(5);
        dump("Running defered code");
    });

    return response('success', 200);
});

```

This code runs perfectly as expected on a production server. I get the response immediately while defered code runs on the background.
On my local machine, it returns the response immediately as well 
**BUT the browser keeps loading until the defered code finishes executing** _after 5 seconds that had been set by `sleep(5)`_

My local OS is **linux debian 12**

What could be causing this problem?

**Comment by ghost** [comment]
Isn't this the same as` \app()->terminating(fn() => ...` ?
see https://dev.to/webong/execute-an-action-after-laravel-returns-response-4pjc

Edit I can't find the function definition in L11.

**Reply by henzeb** [reply]
It runs via a Terminable Middleware. Not sure if app->terminate also works that way or the Terminable Middleware gets executed in app->terminate

**Reply by ghost** [reply]
app terminate is  terminable callback @henzeb but basically should do the same.

**Comment by inmanturbo** [comment]
Are you using `php artisan serve` or some equivalent to serve your project locally? To use defer locally you should use `Valet` (there is a valet linux package) or some other fastcgi, fpm implementation. Defer requires fastcgi. Basically it uses fastcgi to execute on a forked process. Using dev server it would just block until the request is terminated.

**Comment by inmanturbo** [comment]
https://github.com/cpriego/valet-linux

**Comment by inmanturbo** [comment]
Also see https://github.com/cpriego/valet-linux/issues/381
